src/gbm/gblinear.cc

```diff
@@ -69,7 +64,9 @@ class GBLinear : public GradientBooster {
     param_.InitAllowUnknown(cfg);
     updater_.reset(LinearUpdater::Create(param_.updater));
     updater_->Init(cfg);
-    monitor_.Init("GBLinear ", param_.debug_verbose);
+    monitor_.Init(
+        "GBLinear ",
+        ConsoleLogger::GlobalVerbosity() > ConsoleLogger::DefaultVerbosity());
```

Comment:
What about moving this check to inside Monitor::Init() so you don't need the extra argument?